File: TransCoder_tokenized_test_set_functions/python/SUM_NODES_K_TH_LEVEL_TREE_REPRESENTED_STRING_1.py
def sumAtKthLevel ( tree, k, i, level ) :
    if ( tree [ i [ 0 ] ] == '(' ) :
        i [ 0 ] += 1
        if ( tree [ i [ 0 ] ] == ')' ) :
            return 0
        sum = 0
        if ( level == k ) :
            sum = int ( tree [ i [ 0 ] ] )
        i [ 0 ] += 1
        leftsum = sumAtKthLevel ( tree, k, i, level + 1 )
        i [ 0 ] += 1
        rightsum = sumAtKthLevel ( tree, k, i, level + 1 )
        i [ 0 ] += 1
        return sum + leftsum + rightsum
    

File: TransCoder_tokenized_test_set_functions/cpp/SUM_NODES_K_TH_LEVEL_TREE_REPRESENTED_STRING_1.cpp
int sumAtKthLevel ( string tree, int k, int & i, int level ) {
  if ( tree [ i ++ ] == '(' ) {
    if ( tree [ i ] == ')' ) return 0;
    int sum = 0;
    if ( level == k ) sum = tree [ i ] - '0';
    int leftsum = sumAtKthLevel ( tree, k, ++ i, level + 1 );
    int rightsum = sumAtKthLevel ( tree, k, ++ i, level + 1 );
    ++ i;
    return sum + leftsum + rightsum;
  }
}